testapp/configuration.py

```diff
@@ -15,8 +16,17 @@ class UserConfigurationStep(BaseConfigurationStep):
     """
 
     verbose_name = "User Configuration"
-    required_settings = ["USER_CONFIGURATION_USERNAME", "USER_CONFIGURATION_PASSWORD"]
     enable_setting = "USER_CONFIGURATION_ENABLED"
+    required_settings = [
+        "USER_CONFIGURATION_USERNAME",
+        "USER_CONFIGURATION_PASSWORD",
+    ]
+    config_settings = ConfigSettingsModel(
+        models=[User],
+        file_name="user",
+        namespace="USER_CONFIGURATION",
+        excluded_fields=["id", "date_joined", "is_active", "last_login"],
```

Comment:
I'll put here a comment about the implementation in general, because I think this line provides a good example. 
Why do we need to set up `excluded_fields` when we already specify `required_settings` ? I think it's unnecessary. 
With current implementation we need to specify settings twice: as env vars in the configuration step and as model fields in config settings. So it looks like config settings make things not easier, but the opposite. 

Perhaps we have different visions of what the config setting should do. 

I'll specify mine here. We can discuss it together with @stevenbal . 
So I thought config settings class would be used to:
* specify settings used and required for the particular config step
* retrieve help text for them when possible from related model fields
* helps to generate document in .rst for the particular config step

For now config setting class is used to retrieve help test for all model fields of the particular model. 
I don't quite understand why it is needed. Why loop on the all model fields and not on the settings? 
The model can have a lot of fields and we can change only one of them. 

So my main issue is the same as in my previous review - I'd like config-settings to be more coupled with config_step and less with the model.

Reply:
Discussed, @annashamray to provide an example on how to make this better in her eyes, can be discussed with @stevenbal